packages/webdriver/src/bidi/core.ts

```diff
@@ -54,18 +55,52 @@ export class BidiCore {
             return
         }
 
-        this.#waitForConnected = new Promise<boolean>((resolve) => {
-            this.#ws.on('open', () => {
-                log.info('Connected session to Bidi protocol')
+        log.info(`Connecting to webSocketUrl ${this.#webSocketUrl}`)
+        const parsedUrl = new URL(this.#webSocketUrl)
+        // https://github.com/webdriverio/webdriverio/issues/14039
+        const candidateUrls: string[] = [this.#webSocketUrl]
+        if (!isIP(parsedUrl.hostname)) {
+            const [ips4, ips6] = await Promise.all([
+                dns.promises.resolve4(parsedUrl.hostname),
+                dns.promises.resolve6(parsedUrl.hostname),
+            ])
+            const candidateIps = [...ips4, ...ips6]
+            candidateUrls.push(
+                ...candidateIps.map((ip) => this.#webSocketUrl.replace(parsedUrl.hostname, ip))
+            )
```

Comment:
No need to spread everything into the array, concat makes more sense here
```suggestion
            candidateUrls.concat(
                candidateIps.map((ip) => this.#webSocketUrl.replace(parsedUrl.hostname, ip))
            )
```

Reply:
concat does not mutate arrays, but rather returns catenated one. I prefer to mutate the existing array though to reduce the amount of boilerplace code needed to declare intermediate variables